spot-client/src/spot-tv/native-functions/bt-beacon/middleware.js

```diff
@@ -0,0 +1,27 @@
+import { SPOT_TV_SET_JOIN_CODE } from 'common/app-state/spot-tv/action-types';
+import { MiddlewareRegistry } from 'common/redux';
+
+import { nativeController } from '../native-controller';
+
+MiddlewareRegistry.register(() => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case SPOT_TV_SET_JOIN_CODE:
+        _updateJoinCode(action.joinCode);
+    }
+
+    return result;
+});
+
+/**
+ * Updates the native wrapper about a join code change.
+ *
+ * @param {string} joinCode - The join code that the spot TV instance is using now.
```

Comment:
"Spot-TV" is generally how I've been spelling it when I remember I should be consistent.